### ISSUE TITLE
Hang on startup, other oddities

### ISSUE THREAD

**Issue by genericmoniker** [redsolver/noteless#31]
Version 0.1.18

After writing #30, I noticed that the count of notes in the sidebar were not correct, and not all of the Notebooks were showing. I thought I'd restart the app to see if that might update what is shown. The app, upon restart, displayed the blue progress bar under the app title, but didn't progress beyond that for as long as I cared to wait (10 minutes or so?).

If I tap the add note (plus) button, it created a new note and the app started working (breaking out of the the eternal progress bar), but the note counts are still wrong.

I have a little over 400 notes (the app shows 284), and 300-ish attachments. I also have 5 Notebooks (the app shows none) and 39 different tags (the app shows none).

I'm using an external sync mechanism (Syncthing), but all the files appear to be present.

Let me know if there's anything else I can provide to be helpful.

**Comment by redsolver** [comment]
If only a part of your notes is shown and the tags are missing, most likely the listing process was not finished. The tags are determined after every note is loaded.

The app reads the full content of every file to get the metadata, because that was faster than reading and checking every byte of a file (to read only the metadata before the note) when I was testing.
I just tried it in an emulator with 5000 notes(7500 bytes each) and loading took around 1 second.

It would be very helpful if you could try to wait about 30 minutes and look what happens, because that would help to find out if it's a performance problem or if the process crashes somewhere.

**Comment by genericmoniker** [comment]
I left the app running overnight in the background and came back to find it in the same state -- the blue indeterminate progress bar under the title bar was still flowing across the screen. Tapping the plus button doesn't break out anymore.

I also tried uninstalling and reinstalling the app, but I guess there is some state left over between installs because it still failed to start normally. Are there files that I can delete to get back to a clean state?

**Comment by genericmoniker** [comment]
I set up the flutter SDK so I could build and debug the app myself.

The first problem I encountered when starting the app was:

> Exception has occurred.
> FlutterError (ServicesBinding.defaultBinaryMessenger was accessed before the binding was initialized.
> If you're running an application and need to access the binary messenger before `runApp()` has been called (for example, during  plugin initialization), then you need to explicitly call the `WidgetsFlutterBinding.ensureInitialized()` first.
> If you're running a test, you can call the `TestWidgetsFlutterBinding.ensureInitialized()` as the first line in your test's `main()` method to initialize the binding.)

Naively adding a call to `main` seemed to fix it:

```dart
main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await PrefService.init(prefix: 'pref_');
```

The second problem I ran into was when the app was listing my notes. 

> FrontMatterException (FrontMatterException: Front matter is not valid YAML.)

The note in question had a title in quotes, which Notable wrote as:

```yaml
---
title: ""Freezing""
created: '2016-01-20T16:07:34Z'
modified: '2017-06-28T16:49:31Z'
tags: [Notebooks/Software Development, Python]
---
```

If I removed one set of quotes, the app loaded all of my notes.

**Comment by redsolver** [comment]
Which app was used to save this note? Because `title: ""Freezing""` is not valid yaml. Double Quotes should be escaped like this: `
title: "\"Freezing\""`

**Comment by genericmoniker** [comment]
Come to think of it, it was probably my own conversion from OneNote, so I've got a bug. I know I've read that note since the conversion (I guess Notable is tolerant of that problem), but I guess I hadn't edited, at which point Notable would have fixed it.

**Comment by redsolver** [comment]
The upcoming app version will contain some better error handling to help with issues like this: When an exception occurs while reading a note, the note is skipped and the errors are shown as highlighted"virtual notes".

**Comment by redsolver** [comment]
The new error handling is now available in version 0.2.0. If a note can't be parsed, it will be skipped and the new error reports should help identify notes with issues like this.
